addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/studies/StudyPermissionsTable.js

```diff
@@ -133,18 +125,19 @@ class StudyPermissionsTable extends React.Component {
             </Table.Header>
 
             <Table.Body>
-              {this.permissionsStore.studyPermissions.userTypes.map(userType => (
-                <Table.Row key={userType}>
-                  <Table.Cell style={{ textTransform: 'capitalize' }}>{userType}</Table.Cell>
-                  <Table.Cell>
-                    {this.editModeOn ? (
-                      this.renderUsersDropdown(userType)
-                    ) : (
-                      <UserLabels users={this.usersStore.asUserObjects(studyPermissions[`${userType}Users`])} />
-                    )}
-                  </Table.Cell>
-                </Table.Row>
-              ))}
+              {this.permissionsStore.studyPermissions.userTypes.map(userType => {
+                const uids = studyPermissions[`${userType}Users`];
+                const userIdentifiers = _.map(uids, uid => ({ uid }));
+                const users = this.usersStore.asUserObjects(userIdentifiers);
+                return (
```

Comment:
We did not refactor existing code. The existing code was using additional const to make the JSX code read better instead of inlining long expressions in the JSX code below.

Reply:
That's fair, resolving